### PR TITLE
Create pet

### PR DESCRIPTION
**Pull request by Inelukas**

### CODE REVIEW

File: Component/PetSelection/PetSelection.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    &:active {
+      background-color: var(--secondary-color);
+    }
+  }
+`;
+
+const StyledPetIcon = styled.div`
+  width: 120px;
+  height: 120px;
+  display: grid;
```
Comment (P-Oliomonte) [inline Component/PetSelection/PetSelection.js:34]: Why do you use grid here? When the element only contains one element.

File: Component/PetSelection/PetSelection.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      transform: scale(1.2);
+    }
+
+    &:active {
+      background-color: var(--secondary-color);
+    }
+  }
+`;
+
+const StyledPetIcon = styled.div`
```
Comment (P-Oliomonte) [inline Component/PetSelection/PetSelection.js:31]: The PetIcon gets cut off at the top of the screen in mobile view.

File: pages/_app.js [hunk 3]
```diff
@@ -1,10 +1,45 @@
...
+        status: {
+          hunger: 80,
+          happiness: 80,
+          energy: 80,
+          intelligence: 80,
+        },
+      },
+      ...prevData,
+    ]);
+    console.log(petData);
```
Comment (P-Oliomonte) [inline pages/_app.js:31]: Please remove console.log

File: pages/_app.js [hunk 4]
```diff
@@ -1,10 +1,45 @@
...
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
+  const [petCollection, setPetCollection] = useState(pets);
+  const router = useRouter();
+
+  function handleCreatePet(petData) {
+    setPetCollection((prevData) => [
+      {
+        id: uid(),
```
Comment (P-Oliomonte) [inline pages/_app.js:14]: Wouldn't it be easier to create the whole object already inside your form component? 

You accept the petData from your handleSubmit function and create the whole object inside that function.

File: pages/create/index.js [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  gap: 30px;
+  background-image: var(--create-image);
+`;
+
+const StyledConfirmButtonContainer = styled.section`
+  display: flex;
+`;
+
+const StyledReturnButton = styled.button`
+  display: grid;
```
Comment (P-Oliomonte) [inline pages/create/index.js:24]: See above. Why grid?

File: pages/create/index.js [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+  }
+
+  label {
+    width: 20%;
+    border: none;
+  }
+`;
+
+export default function CreatePetPage({ onCreatePet }) {
+  const [currentPet, setCurrentPet] = useState(0);
```
Comment (P-Oliomonte) [inline pages/create/index.js:115]: You shouldn't use a numeric number as default value when later used for objects. If you want to use an empty default value use `null`

File: pages/create/index.js [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+    setCurrentPet(
+      direction === "back"
+        ? currentPet === 0
+          ? animalList.length - 1
+          : currentPet - 1
+        : currentPet === animalList.length - 1
+        ? 0
+        : currentPet + 1
+    );
+  }
```
Comment (P-Oliomonte) [inline pages/create/index.js:132]: The ternaries are very nested and hard to follow. For better readability maybe write this function like this: 

```suggestion
 function handleChangePet(direction) {
  let newPetIndex;

  direction === "back" && currentPet === 0 && (newPetIndex = animalList.length - 1);
  direction === "back" && currentPet !== 0 && (newPetIndex = currentPet - 1);
  direction !== "back" && currentPet === animalList.length - 1 && (newPetIndex = 0);
  direction !== "back" && currentPet !== animalList.length - 1 && (newPetIndex = currentPet + 1);

  setCurrentPet(newPetIndex);
}```

File: pages/create/index.js [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+      <PetSelection
+        onChangePet={handleChangePet}
+        animalList={animalList}
+        currentPet={currentPet}
+        name="type"
+      />
+      <StyledMainField>
+        <StyledFormElement>
+          <label>Type:</label>
+          <input name="type" value={animalList[currentPet].type} disabled />
```
Comment (P-Oliomonte) [inline pages/create/index.js:162]: Here you missed to include `htmlFor` and `id`

File: pages/create/index.js [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+        </StyledFormElement>
+      </StyledMainField>
+      <StyledIndicatorContainer>
+        {animalList[currentPet].indicators.map((indicator, index) => (
+          <Indicator key={index} data={indicator} />
+        ))}
+      </StyledIndicatorContainer>
+      <StyledConfirmButtonContainer>
+        <StyledReturnButton type="button">
+          <Link href="/">Cancel</Link>
```
Comment (P-Oliomonte) [inline pages/create/index.js:237]: See above. Don't put Links inside buttons. Furthermore for a Cancel button we would prefer to use a button not a Link.

**Review by P-Oliomonte — CHANGES_REQUESTED** [review]
Hi Team! 

Good start! We have some suggestions: 

Maybe you should leave out the media queries for now and concentrate on mobile first. A lot of things overlap the container elements in the mobile view. Take another look at that. With mobile first, you wouldn't need hovers and pointers and so on.

Also, maybe rethink your HTML elements and make sure to use more semantic HTML, that's important for accessibility. 

But overall: Everything works as it should and looks fun. Thumbs up from us 👍

File: Component/PetList/PetList.js [hunk 10]
```diff
@@ -1,20 +1,52 @@
...
   flex-direction: column;
   align-items: center;
   width: 100vw;
+  margin: 5% 0;
 `;
 
-export default function PetList() {
-  return pets.length > 0 ? (
+const StyledCreateButton = styled.button`
+  display: grid;
```
[the remark this answers is not in the document]
Comment (Inelukas) [reply Component/PetList/PetList.js:14]: I usually use "display: grid; place-content: center" to center an element, in this case the text content of the button. Is there another easier way zo achieve this?

File: pages/create/index.js (hunk 6 above)
Comment (Inelukas) [reply pages/create/index.js:115]: currentPet is currently only the indicator for the index of a pet object inside an array, not the object itself

File: pages/create/index.js (hunk 8 above)
Comment (Inelukas) [reply pages/create/index.js:162]: is this really necessary if the element is always disabled anyways?

File: pages/create/index.js (hunk 9 above)
Comment (Inelukas) [reply pages/create/index.js:237]: Not sure what exactly you mean here. So you would suggest using a button and then use route.push inside a separate function that is triggered upon clicking the button?

File: Component/PetList/PetList.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (P-Oliomonte) [reply Component/PetList/PetList.js:14]: Here I would just make a Link without the button as a wrapper and style the Link itself accordingly. 

  display: flex;
  align-items: center;
  justify-content: center;

File: pages/create/index.js (hunk 6 above)
Comment (P-Oliomonte) [reply pages/create/index.js:115]: Ah, alright! That makes sense. Ignore our comment then.

File: pages/create/index.js (hunk 8 above)
Comment (P-Oliomonte) [reply pages/create/index.js:162]: I think semantically and for accessibility purposes it always makes sense to pair the two elements.

File: pages/create/index.js (hunk 9 above)
Comment (P-Oliomonte) [reply pages/create/index.js:237]: Sorry, the other way around. Use a Link and style it accordingly. But don't wrap a Link inside a button. Use one element or the other.

File: pages/create/index.js [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+    font-family: sans-serif;
+    border: 1px solid #000000;
+
+    &:hover {
+      transform: scale(1.2);
+    }
+
+    &:active {
+      background-color: var(--secondary-color);
+    }
```
Comment (NikRoe) [inline pages/create/index.js:54]: This anchor tag has the same styling as the link on your starting page.

Make this a StyledComponent as use it in two places

File: pages/create/index.js [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+  box-shadow: 2px 2px black;
+  cursor: pointer;
+  background-color: var(--signal-color);
+  background-image: var(--button-image);
+  border: 1px solid #000000;
+
+  a {
+    color: #000000;
+    text-decoration: none;
+  }
```
Comment (NikRoe) [inline pages/create/index.js:76]: You should never nest an anchor inside a button or vice versa

File: pages/create/index.js [hunk 13]
```diff
@@ -0,0 +1,311 @@
...
+                      characteristics.characteristic2 ===
+                        option.characteristic ||
+                      characteristics.characteristic2 === option.opposite
+                    }
+                  >
+                    {option.characteristic}
+                  </option>
+                ))}
+              </select>
+              <select
```
Comment (NikRoe) [inline pages/create/index.js:272]: this form element does not have a label it is connected to, making the form less accessible

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

I left some comments for you to take a look at.

They cover semantic HTML, Styled Components and simplification of your code.

There are also some comments not yet resolved from the previous review, concerning console.logs etc. Please also look into those comments and take them into account.

Keep it up 🙌

File: Component/PetList/PetList.js [hunk 14]
```diff
@@ -1,20 +1,48 @@
...
+    color: #000000;
+    text-decoration: none;
+
+    &:hover {
+      transform: scale(1.2);
+    }
+
+    &:active {
+      background-color: var(--secondary-color);
+    }
```
[the remark this answers is not in the document]
Comment (Inelukas) [reply Component/PetList/PetList.js:34]: do I also have to avoid nested styles inside for example a button or Link component? I created a Link component but as I have to put the Link element inside another element (as I cannot declare a "styled.Link") I don't now how I could avoid nesting the styles for the Link element inside its parent.
Also, do pseudo classes count as nested classes? If yes, how can I implement these features without nesting?

File: pages/create/index.js
[the remark this answers is not in the document]
Comment (Inelukas) [reply pages/create/index.js:1]: about the sections comment: These were originally divs but I was told I shouldn't use too many divs. What could I choose instead? Maybe article elements?

[the remark this answers is not in the document]
Comment (Inelukas) [reply pages/create/index.js:1]: regarding the form restructuring: I tried both of your approaches but none of them seems to be working. Now, nothing happens upon clicking the button and I have no idea how to solve this issue. AI is not being of any help either.
Is it really necessary to restructure the code as you said? For me, the code is much more comprehendible if I just put all necessary elements inside the form

File: Component/PetList/PetList.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (NikRoe) [reply Component/PetList/PetList.js:34]: You can extend the Link component using Styled Components:

```js
const StyledLink = styled(Link)``;
```

Pseudo classes don't count as nested classes. They need to be part of a parent, so that there is a reference.

File: pages/create/index.js
[the remark this answers is not in the document]
Comment (NikRoe) [reply pages/create/index.js:1]: Yes, you should refactor your code to have a different HTML structure. Since the HTML gives your page its structure and logic. What is relevant for the form should be part of the form element. 
The other HTML you try to display is not related to the form element, therefore, they should be outside of the form.

In general it does not make much sense to have 20 div elements or 20 section elements in your application.
Each element has its own purpose and should be used accordingly.

If it is something independent and reusable, an [article](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article) might make sense.
If is a bigger blog where there is a heading, a [section](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section) might make sense.
If you only need a block container for styling, a div makes sense.

Regarding your current issue with the submission of the form: Please push all current changes on the branch, create a discussion on the [board](https://github.com/orgs/neuefische/discussions), link the current PR and describe what you tried, what works, whats does not. I will take a look at it and we will fix it.

File: Component/PetList/PetList.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (Inelukas) [reply Component/PetList/PetList.js:34]: Ok cool, that helped a lot! 😇

File: pages/create/index.js
[the remark this answers is not in the document]
Comment (Inelukas) [reply pages/create/index.js:1]: Actuallly I just solved it. I just forgot to add an addClick event to the button component I created 😅 Thanks so much for your help!

File: pages/_app.js [hunk 15]
```diff
@@ -1,10 +1,44 @@
...
+        name: petData.name,
```
Comment (Roland-Hufnagel) [inline pages/_app.js:15]: I agree to P-Oliomonte: 
since you get the petData as an argument you can enrich it here like:
```js
//...
{ ...petData,
  id: uid()
}
```
Also you add the status to the petData inside the handleSubmit. At the moment in the form are different values for hunger, happiness, etc displayed but you give them all 80% here. I think that's not what you want. You shoud read the values inside the handlesubmit and add it to the petData.

File: pages/_app.js [hunk 16]
```diff
@@ -1,10 +1,44 @@
...
+        type: petData.type,
+        picture: petData.picture,
+        characteristics: [
+          petData.characteristic1,
+          petData.characteristic2,
+        ].filter(Boolean),
```
Comment (Roland-Hufnagel) [inline pages/_app.js:21]: I'm not sure but I think you don't need that.

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hi Team. You app looks really nice 👍. But the code can be improved a lot. There are a few minor important things but also a few important ones. Keep in mind: 'Keep it simple'. The better the structure and the cleaner the code right from the beginning the less problems you will have in the future.

File: pages/_app.js (hunk 16 above)
Comment (Inelukas) [reply pages/_app.js:21]: this is currently necessary as the second characteristic is optional and should only be included in the object if its value is not an empty string

File: pages/create/index.js [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+  return (
+    <StyledCreatePage>
+      <PetSelection
+        onPreviousPet={handlePreviousPet}
+        onNextPet={handleNextPet}
+        animalList={animalList}
+        currentPet={currentPet}
+        name="type"
+      />
+      <StyledForm ref={formRef} onSubmit={handleSubmit}>
```
[the remark this answers is not in the document]
Comment (Inelukas) [reply pages/create/index.js]: I created a new component but now the code has become much more complicated because I had to implement ref and forwardRef. It will take me some time to figure out how that works 😅

[the remark this answers is not in the document]
Comment (Roland-Hufnagel) [reply pages/create/index.js]: You can create a PetForm component. That component can contain all the stuff you need there. Including the useRef, the button, etc. 
You only have to be aware not to include unnecessary  elements inside the `<form></form>` element itself.

**Review by Roland-Hufnagel — APPROVED** [review]
Looks good now 👍
